reflect.go

```diff
@@ -0,0 +1,388 @@
+package main
+
+import (
+	"fmt"
+	"go/types"
+	"path/filepath"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/tools/go/ssa"
+)
+
+var checkedAPIs = make(map[string]bool)
+
+// Record all instances of reflection use, and don't obfuscate types which are used in reflection.
+func (tf *transformer) recordReflection(ssaPkg *ssa.Package) {
+	lenPrevKnownReflectAPIs := len(cachedOutput.KnownReflectAPIs)
+
+	notCheckedAPIs := make(map[string]bool)
```

Comment:
can you document this checked and not checked map logic a bit? it's not immediately obvious to me

Reply:
There is a visited and notVisited map, at the first pass all knowReflectAPIs are loaded into the notVisited map, on subsequent passes only knowReflectAPIs \ visited.
After the pass we know that all APIs in the notVisited map have been explored, they are then saved to the visited map to avoid checking the same API multiple times.

Otherwise we would do the same work over and over again, when a new knowReflectAPI is found now only that new one is tested.

Reply:
Will add docs